tests/test_socket_server.py

```diff
@@ -85,15 +81,22 @@ async def write(self, **data):
             The data to write.
         """
         st = json.dumps({**data})
-        self.writer.write(st.encode() + b"\r\n")
-        self.log.debug(st)
+        self.writer.write(st.encode() + tcpip.TERMINATOR)
         await self.writer.drain()
 
+    async def test_disconnect(self):
+        self.assertTrue(self.srv.connected)
+        await self.write(command="disconnect", parameters={})
+        # Give time to the socket server to clean up internal state and exit.
```

Comment:
You can do this, instead: `await self.srv.done_task` to be more efficient (most of the time) and avoid a race condition if Jenkins is very badly bogged down.

Reply:
I tried and this makes the unit test hang so I'll leave it as is.